rasa_core/training/generator.py

```diff
@@ -266,15 +348,52 @@ def _remove_duplicate_trackers(self, trackers):
         same featurization once will do so in the future (as we
         feed the same events to all trackers)."""
 
+        step_hashed_featurizations = set()
+
         # collected trackers that created different featurizations
-        unique_trackers = []
+        unique_trackers = []  # for current step
+        end_trackers = []  # for all steps
 
         for tracker in trackers:
             states = self.domain.states_for_tracker_history(tracker)
             hashed = hash(tuple((frozenset(s) for s in states)))
```

Comment:
if it is not too much trouble, I guess let's add them, since we have them present there. Right now I think the value is 1 if it is present, except for slots probably.